### PR TITLE
[Metrics Advisor] Add remaining docstrings

### PR DESCRIPTION
**Pull request by christothes**

### CODE REVIEW

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedAccessMode.cs [hunk 1]
```diff
@@ -5,17 +5,19 @@
...
 {
-    // TODODOCS.
     /// <summary>
+    /// The view mode of the <see cref="DataFeed"/>.
     /// </summary>
     [CodeGenModel("ViewMode")]
     public readonly partial struct DataFeedAccessMode
     {
         /// <summary>
+        /// Indicates that the view data feed is private.
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedAccessMode.cs:15]:
```suggestion
        /// Indicates that the data feed is private.
```

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedAutoRollupMethod.cs [hunk 2]
```diff
@@ -6,13 +6,14 @@
 namespace Azure.AI.MetricsAdvisor.Models
 {
     /// <summary>
-    /// The roll-up method the service should apply to the ingested data for anomaly detection.
+    /// The roll-up method perform aggregation (for example SUM, MAX, MIN) on each dimension during ingestion used to
+    /// build a hierarchy which will be used in root case analysis and other diagnostic features.
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedAutoRollupMethod.cs:10]: Should it be:
> ...on each dimension during ingestion. Used to build...
?

I'm having trouble parsing the whole sentence.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedIngestionSettings.cs [hunk 3]
```diff
@@ -61,8 +60,8 @@ internal DataFeedIngestionSettings(DataFeedDetail dataFeedDetail)
         /// </summary>
         public TimeSpan? IngestionStartOffset { get; set; }
 
-        // TODODOCS.
         /// <summary>
+        /// Specifies the maximum amount of time in which to attempt a retry if ingestions fails.
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedIngestionSettings.cs:64]:
```suggestion
        /// Specifies the maximum amount of time in which to attempt a retry if ingestion fails.
```

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/GetDataFeedIngestionStatusesOptions.cs [hunk 4]
```diff
@@ -2,34 +2,44 @@
 // Licensed under the MIT License.
 
 using System;
+using Azure.AI.MetricsAdvisor.Administration;
 
 namespace Azure.AI.MetricsAdvisor.Models
 {
     /// <summary>
+    /// The set of options that can be specified when calling <see cref="MetricsAdvisorAdministrationClient.GetDataFeedIngestionProgress"/> or
+    /// <see cref="MetricsAdvisorAdministrationClient.GetDataFeedIngestionProgressAsync"/> to configure the behavior of the request.
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/GetDataFeedIngestionStatusesOptions.cs:11]:
```suggestion
    /// The set of options that can be specified when calling <see cref="MetricsAdvisorAdministrationClient.GetDataFeedIngestionStatuses"/> or
    /// <see cref="MetricsAdvisorAdministrationClient.GetDataFeedIngestionStatusesAsync"/> to configure the behavior of the request.
```

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs [hunk 5]
```diff
@@ -572,9 +582,13 @@ public virtual Response RefreshDataFeedIngestion(string dataFeedId, DateTimeOffs
...
         }
 
-        // TODODOCS.
         /// <summary>
+        /// Gets the ingestion status for data being ingested to a given data feed.
         /// </summary>
+        /// <param name="dataFeedId">The unique identifier of the <see cref="DataFeed"/>.</param>
+        /// <param name="options">An optional set of options used to configure the request's behavior.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>An <see cref="AsyncPageable{T}"/> containing the collection of <see cref="DataFeedIngestionStatus"/>s.</returns>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs:591]: nit: Statuss

Should we leave only as `Status`? Or would `Statuses` be better?

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs [hunk 6]
```diff
@@ -621,9 +635,13 @@ async Task<Page<DataFeedIngestionStatus>> NextPageFunc(string nextLink, int? pag
...
         }
 
-        // TODODOCS.
         /// <summary>
+        /// Gets the ingestion status for data being ingested to a given data feed.
         /// </summary>
+        /// <param name="dataFeedId">The unique identifier of the <see cref="DataFeed"/>.</param>
+        /// <param name="options">An optional set of options used to configure the request's behavior.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>An <see cref="Pageable{T}"/> containing the collection of <see cref="DataFeedIngestionStatus"/>s.</returns>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs:644]: nit: Statuss

Should we leave only as `Status`? Or would `Statuses` be better?

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs [hunk 7]
```diff
@@ -1013,9 +1013,13 @@ Page<AnomalyIncident> NextPageFunc(string nextLink, int? pageSizeHint)
             return PageableHelpers.CreateEnumerable(FirstPageFunc, NextPageFunc);
         }
 
-        // TODODOCS.
         /// <summary>
+        /// Gets the automatic suggestions for likely root causes of an incident.
         /// </summary>
+        /// <param name="detectionConfigurationId">The unique identifier of the <see cref="MetricAnomalyAlertConfiguration"/>.</param>
+        /// <param name="incidentId">The unique identifier of the <see cref="AnomalyIncident"/>.</param>
+        /// <param name="cancellationToken"></param>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:1021]: Missing.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs [hunk 8]
```diff
@@ -1039,7 +1043,12 @@ public virtual async Task<Response<IReadOnlyList<IncidentRootCause>>> GetInciden
         }
 
         /// <summary>
+        /// Gets the automatic suggestions for likely root causes of an incident.
         /// </summary>
+        /// <param name="detectionConfigurationId">The unique identifier of the <see cref="MetricAnomalyAlertConfiguration"/>.</param>
+        /// <param name="incidentId">The unique identifier of the <see cref="AnomalyIncident"/>.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:1051]: Missing both.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs [hunk 9]
```diff
@@ -1223,7 +1232,17 @@ public virtual async Task<Response<IReadOnlyList<MetricEnrichedSeriesData>>> Get
...
         /// </summary>
+        /// <param name="seriesKeys">The name of the dimension.</param>
+        /// <param name="detectionConfigurationId">The unique identifier of the <see cref="MetricAnomalyAlertConfiguration"/>.</param>
+        /// <param name="startTime">Filters the result. Only data points ingested after this point in time, in UTC, will be returned.</param>
+        /// <param name="endTime">Filters the result. Only data points ingested after this point in time, in UTC, will be returned.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>A collection containing the values the specified dimension assumed for anomalous data points. Items are unique.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="seriesKeys"/> or <paramref name="detectionConfigurationId"/> is null.</exception>
+        /// <exception cref="ArgumentException"><paramref name="seriesKeys"/> or <paramref name="detectionConfigurationId"/> is empty; or <paramref name="detectionConfigurationId"/> is not a valid GUID.</exception>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:1245]: This description is very inaccurate. I think you copied it by accident from one the methods marked with TODODOCS (probably GetMetricEnrichedSeriesDataAsync).

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AnomalyAlert/TopNGroupScope.cs [hunk 10]
```diff
@@ -27,14 +29,17 @@ public TopNGroupScope(int top, int period, int minimumTopCount)
...
         /// </summary>
         public int Top { get; }
 
         /// <summary>
+        /// The number of items a period contains.
         /// </summary>
         public int Period { get; }
 
         /// <summary>
+        /// Th number of anomalies that must be in the specified <see cref="Top"/> number of timestamps to send an alert.
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AnomalyAlert/TopNGroupScope.cs:42]:
```suggestion
        /// The number of anomalies that must be in the specified <see cref="Top"/> number of timestamps to send an alert.
```

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AnomalyDetection/ChangeThresholdCondition.cs [hunk 11]
```diff
@@ -19,10 +17,10 @@ public partial class ChangeThresholdCondition
         /// <summary>
         /// Initializes a new instance of the <see cref="ChangeThresholdCondition"/> class.
         /// </summary>
-        /// <param name="changePercentage">The relative change. </param>
+        /// <param name="changePercentage">Compared to the previous point, the current point is an anomaly if the percentage of change is more than this value. </param>
         /// <param name="shiftPoint">When set to N, sets as reference the data point that's N positions before the current point. Value must be at least 1.</param>
-        /// <param name="isWithinRange"></param>
-        /// <param name="anomalyDetectorDirection"></param>
+        /// <param name="isWithinRange">Sets whether to detect data within the range or outsdie the range.</param>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AnomalyDetection/ChangeThresholdCondition.cs:22]:
```suggestion
        /// <param name="isWithinRange">Sets whether to detect data within the range or outside the range.</param>
```

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AnomalyDetection/ChangeThresholdCondition.cs [hunk 12]
```diff
@@ -43,15 +41,19 @@ public ChangeThresholdCondition(double changePercentage, int shiftPoint, bool is
         public int ShiftPoint { get; }
 
         /// <summary>
+        /// Compared to the previous point, the current point is an anomaly if the percentage of change is more than this value.
         /// </summary>
         public double ChangePercentage { get; }
 
         /// <summary>
+        /// Sets whether to detect data within the range or outsdie the range. If you want to detect fluctuations, select false.
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AnomalyDetection/ChangeThresholdCondition.cs:49]:
```suggestion
        /// Sets whether to detect data within the range or outside the range. If you want to detect fluctuations, select false.
```

**Review by kinelski — COMMENTED** [review]
Left some minor nit comments.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs (hunk 5 above)
Comment (christothes) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs:591]: I'll leave it as Status for now.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs (hunk 9 above)
Comment (christothes) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:1245]: Not sure how I missed this, thanks.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs [hunk 13]
```diff
@@ -1013,9 +1013,15 @@ Page<AnomalyIncident> NextPageFunc(string nextLink, int? pageSizeHint)
             return PageableHelpers.CreateEnumerable(FirstPageFunc, NextPageFunc);
         }
 
-        // TODODOCS.
         /// <summary>
+        /// Gets the automatic suggestions for likely root causes of an incident.
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:1017]: what does automatic suggestions means? what will be a "manual" suggestion?

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs [hunk 14]
```diff
@@ -1187,15 +1200,14 @@ Page<string> NextPageFunc(string nextLink, int? pageSizeHint)
         }
 
         /// <summary>
-        /// Gets data about the data points present in the specified set of time series.
-        /// <see cref="MetricAnomalyAlertConfiguration"/>.
+        /// Query series enriched by anomaly detection.
         /// </summary>
-        /// <param name="seriesKeys">The name of the dimension.</param>
+        /// <param name="seriesKeys">The deection series keys.</param>
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:1205]:
```suggestion
        /// <param name="seriesKeys">The detection series keys.</param>
```

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedAccessMode.cs [hunk 15]
```diff
@@ -5,17 +5,19 @@
...
 {
-    // TODODOCS.
     /// <summary>
+    /// The view mode of the <see cref="DataFeed"/>.
     /// </summary>
     [CodeGenModel("ViewMode")]
     public readonly partial struct DataFeedAccessMode
     {
         /// <summary>
+        /// Indicates that the data feed is private.
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedAccessMode.cs:15]: This could be me missing content, but private to whom?

Comment (maririos) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedAccessMode.cs:15]: only admins can see it?

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedIngestionSettings.cs [hunk 16]
```diff
@@ -46,11 +46,10 @@ internal DataFeedIngestionSettings(DataFeedDetail dataFeedDetail)
         /// </summary>
         public int? DataSourceRequestConcurrency { get; set; }
 
-        // TODO: provide link to documentation in IngestionRetryDelay.
-
         /// <summary>
         /// The minimum delay between two consecutive retry attempts, in case data ingestion
         /// fails. If not specified, the service's behavior depends on the data feed's granularity.
+        /// See the <see href="https://docs.microsoft.com/en-us/azure/cognitive-services/metrics-advisor/how-tos/onboard-your-data#avoid-loading-partial-data">documentation></see> for details.
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedIngestionSettings.cs:52]: could you remove locale from the link?

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedOptions.cs [hunk 17]
```diff
@@ -42,13 +42,15 @@ internal DataFeedOptions(DataFeedDetail dataFeedDetail)
         /// </summary>
         public string Description { get; set; }
 
-        // TODODOCS.
         /// <summary>
+        /// Defines actionable HTTP urls, which consist of the placeholders %datafeed, %metric, %timestamp, %detect_config, and %tagset.
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedOptions.cs:46]:
```suggestion
        /// Defines actionable HTTP URLs, which consist of the placeholders %datafeed, %metric, %timestamp, %detect_config, and %tagset.
```

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedOptions.cs [hunk 18]
```diff
@@ -42,13 +42,15 @@ internal DataFeedOptions(DataFeedDetail dataFeedDetail)
...
+        /// You can use the template to redirect from an anomaly or an incident to a specific URL to drill down.
+        /// See the <see href="https://docs.microsoft.com/en-us/azure/cognitive-services/metrics-advisor/how-tos/manage-data-feeds#action-link-template">documentation</see> for details.
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedOptions.cs:48]:
```suggestion
        /// See the <see href="https://docs.microsoft.com/azure/cognitive-services/metrics-advisor/how-tos/manage-data-feeds#action-link-template">documentation</see> for details.
```

**Comment by maririos** [comment]
for all the TODODOCS that had questions about the service behavior, where the questions answered?

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs [hunk 19]
```diff
@@ -1013,9 +1013,15 @@ Page<AnomalyIncident> NextPageFunc(string nextLink, int? pageSizeHint)
...
 
-        // TODODOCS.
         /// <summary>
+        /// Gets the automatic suggestions for likely root causes of an incident.
         /// </summary>
+        /// <param name="detectionConfigurationId">The unique identifier of the <see cref="MetricAnomalyAlertConfiguration"/>.</param>
+        /// <param name="incidentId">The unique identifier of the <see cref="AnomalyIncident"/>.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>A collection of <see cref="IncidentRootCause"/>s for the specified alert configuration and incident.</returns>
+        /// /// <exception cref="ArgumentNullException"><paramref name="detectionConfigurationId"/> or <paramref name="incidentId"/> is null.</exception>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:1023]: nit: extra `///`

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs [hunk 20]
```diff
@@ -1039,7 +1045,14 @@ public virtual async Task<Response<IReadOnlyList<IncidentRootCause>>> GetInciden
         }
 
         /// <summary>
+        /// Gets the automatic suggestions for likely root causes of an incident.
         /// </summary>
+        /// <param name="detectionConfigurationId">The unique identifier of the <see cref="MetricAnomalyAlertConfiguration"/>.</param>
+        /// <param name="incidentId">The unique identifier of the <see cref="AnomalyIncident"/>.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>A collection of <see cref="IncidentRootCause"/>s for the specified alert configuration and incident.</returns>
+        /// /// <exception cref="ArgumentNullException"><paramref name="detectionConfigurationId"/> or <paramref name="incidentId"/> is null.</exception>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:1054]: nit: extra `///`

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs [hunk 21]
```diff
@@ -1187,15 +1200,14 @@ Page<string> NextPageFunc(string nextLink, int? pageSizeHint)
...
-        /// <param name="seriesKeys">The name of the dimension.</param>
+        /// <param name="seriesKeys">The deection series keys.</param>
         /// <param name="detectionConfigurationId">The unique identifier of the <see cref="MetricAnomalyAlertConfiguration"/>.</param>
-        /// <param name="startTime">Filters the result. Only data points ingested after this point in time, in UTC, will be returned.</param>
-        /// <param name="endTime">Filters the result. Only data points ingested after this point in time, in UTC, will be returned.</param>
+        /// <param name="startTime">Filters the result. Only data points after this point in time, in UTC, will be returned.</param>
+        /// <param name="endTime">Filters the result. Only data points after this point in time, in UTC, will be returned.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
-        /// <returns>A collection containing the values the specified dimension assumed for anomalous data points. Items are unique.</returns>
+        /// <returns>A collection <see cref="MetricEnrichedSeriesData"/>.</returns>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:1210]: Maybe a collection "of"?

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs [hunk 22]
```diff
@@ -1223,7 +1235,16 @@ public virtual async Task<Response<IReadOnlyList<MetricEnrichedSeriesData>>> Get
...
 
         /// <summary>
+        /// Query series enriched by anomaly detection.
         /// </summary>
+        /// <param name="seriesKeys">The deection series keys.</param>
+        /// <param name="detectionConfigurationId">The unique identifier of the <see cref="MetricAnomalyAlertConfiguration"/>.</param>
+        /// <param name="startTime">Filters the result. Only data points after this point in time, in UTC, will be returned.</param>
+        /// <param name="endTime">Filters the result. Only data points after this point in time, in UTC, will be returned.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>A collection <see cref="MetricEnrichedSeriesData"/>.</returns>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:1245]: Same here. "of"?

**Review by kinelski — APPROVED** [review]
LGTM. Left some nit picky comments.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs (hunk 13 above)
Comment (christothes) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:1017]: I guess all suggestions are automatic :)

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedAccessMode.cs (hunk 15 above)
Comment (christothes) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedAccessMode.cs:15]: We need to clarify this.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedIngestionSettings.cs (hunk 16 above)
Comment (christothes) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedIngestionSettings.cs:52]: thanks - I always forget that - we need an analyzer for this :)

Comment (maririos) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedIngestionSettings.cs]: me too. I think eng services added a check in the CI for that. OR they are planning on it

**Review by maririos — APPROVED** [review]

**Pull request merged**